HTCONDOR-740 - don't mess with user-held jobs

JaimeFrey opened the pull request:
https://opensciencegrid.atlassian.net/browse/HTCONDOR-740

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

johnkn commented on src/condor_schedd.V6/schedd.cpp:
```diff
@@ -3873,7 +3873,8 @@ ResponsibleForPeriodicExprs( JobQueueJob *jobad, int & status )
 		jobad->LookupInteger(ATTR_HOLD_REASON_CODE,hold_reason_code);
-		if( hold_reason_code == CONDOR_HOLD_CODE::SpoolingInput ) {
+		if( hold_reason_code == CONDOR_HOLD_CODE::SpoolingInput ||
+		    hold_reason_code == CONDOR_HOLD_CODE::UserRequest ) {
```
do we need to knob this?

JaimeFrey replied:
I don't think we need a knob.

johnkn reviewed (changes requested):
The changes to AnalyzePolicy need some more thought as to how release interacts with other policies.

JaimeFrey commented:
I fixed the issues identified by TJ and the unit tests. I also removed one redundant lookup of JobStatus when the schedd and gridmanager do periodic eval of all of their jobs.

johnkn reviewed (approved):
Looks good
